src/sparse_blas/backends/cusparse/operations/cusparse_spmv.cpp

```diff
@@ -0,0 +1,302 @@
+                         is_beta_host_accessible, y_handle, alg, spmv_descr);
+    spmv_descr->workspace.usm_ptr = workspace;
+    if (alg == spmv_alg::no_optimize_alg) {
+        return detail::collapse_dependencies(queue, dependencies);
+    }
+
+#if CUSPARSE_VERSION < 12300
+    // cusparseSpMV_preprocess was added in cuSPARSE 12.3.0.142 (CUDA 12.4)
+    return detail::collapse_dependencies(queue, dependencies);
+#else
+    auto functor = [=](CusparseScopedContextHandler &sc) {
+        auto cu_handle = sc.get_handle(queue);
+        spmv_optimize_impl(cu_handle, opA, alpha, A_handle, x_handle, beta, y_handle, alg,
+                           workspace, is_alpha_host_accessible);
+    };
+    return dispatch_submit(__func__, queue, dependencies, functor, A_handle, x_handle, y_handle);
+#endif
+}
+
+sycl::event spmv(sycl::queue &queue, oneapi::mkl::transpose opA, const void *alpha,
+                 matrix_view A_view, matrix_handle_t A_handle, dense_vector_handle_t x_handle,
+                 const void *beta, dense_vector_handle_t y_handle, spmv_alg alg,
+                 spmv_descr_t spmv_descr, const std::vector<sycl::event> &dependencies) {
+    bool is_alpha_host_accessible = detail::is_ptr_accessible_on_host(queue, alpha);
+    bool is_beta_host_accessible = detail::is_ptr_accessible_on_host(queue, beta);
+    check_valid_spmv(__func__, opA, A_view, A_handle, x_handle, y_handle, is_alpha_host_accessible,
+                     is_beta_host_accessible);
+    if (A_handle->all_use_buffer() != spmv_descr->workspace.use_buffer()) {
+        detail::throw_incompatible_container(__func__);
+    }
+
+    if (!spmv_descr->optimized_called) {
+        throw mkl::uninitialized("sparse_blas", __func__,
+                                 "spmv_optimize must be called before spmv.");
+    }
+    CHECK_DESCR_MATCH(spmv_descr, opA, "spmv_optimize");
+    CHECK_DESCR_MATCH(spmv_descr, A_view, "spmv_optimize");
+    CHECK_DESCR_MATCH(spmv_descr, A_handle, "spmv_optimize");
+    CHECK_DESCR_MATCH(spmv_descr, x_handle, "spmv_optimize");
+    CHECK_DESCR_MATCH(spmv_descr, y_handle, "spmv_optimize");
+    CHECK_DESCR_MATCH(spmv_descr, alg, "spmv_optimize");
+
+    auto compute_functor = [=](CusparseScopedContextHandler &sc, void *workspace_ptr) {
+        auto [cu_handle, cu_stream] = sc.get_handle_and_stream(queue);
+        auto cu_a = A_handle->backend_handle;
+        auto cu_x = x_handle->backend_handle;
+        auto cu_y = y_handle->backend_handle;
+        auto type = A_handle->value_container.data_type;
+        auto cu_op = get_cuda_operation(type, opA);
+        auto cu_type = get_cuda_value_type(type);
+        auto cu_alg = get_cuda_spmv_alg(alg);
+        // Workaround issue with captured alpha and beta causing a segfault inside cuSPARSE
+        // Copy alpha and beta locally in the largest data value type and use the local pointer
+        cuDoubleComplex local_alpha, local_beta;
```

Comment:
Why is there a `cuDoubleComplex` coming into the picture here? When did you implement this workaround? If it has been a while, then with all the changes that have gone in in the last month, can you try removing this workaround for `alpha`/`beta` and see if things are working normally?

Reply:
You didn't need this in `spmm` for instance.

Reply:
This was needed in the first version. Thanks for pointing it out, it seems it is not needed anymore! This is removed in https://github.com/oneapi-src/oneMKL/pull/527/commits/6946d64b3bdba86ecc456d5b2c8d15f945dc5f09